free-1.md

```diff
@@ -9,42 +9,141 @@ good way to meet their needs.
+inefficient to continue doing.  So, I thought having a domain specific language
+for this would be a great solution. 
 
 ### Domain
 _Describe the project's domain in five words._
-
+Creating graphs for economics notes
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
-
+The user would likely be using some sort of text editor or word processor for
+taking notes. This is an early problem for me: can I create the language to be
+usable in popular word processors? Regardless, I would imagine some note snippets
+to look similar to the following in this language:
+
+``` 
+This is a standard supply and demand graph
+
+Graph{
+	Y-Axis: Price
+	X-Axis: Quantity
+	SupplyCurve()	
+	DemandCurve()
+}
+
+generate Graph
+```
+
+```
+Today, we're learning about the *Engel Curve*, the graph looks like this: 
+
+Graph{
+	Y-axis: Income/month
+	X-axis: Units
+	EngelCurve()
+}
+
+generate Graph
+```	
+
+By instantiating some Graph object, you can generate a graph in place in the 
+notes. Then the user could label axes and specify the curves that are needed 
+in the graph. So you could specify SupplyCurve(), which would give a default
+upward sloping supply curve. Each curve could take certain parameters for
+transformations, for example `SupplyCurve(Slope: .5, Y: -10, X: 50  )`, would
+give you a supply curve with slope of .5 starting at a Y-value of -10 and X-value
+of 50. Also, we'd want to support markdown in the language to give users
+easy formatting options when writing notes.
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
+The program would ideally run on save.  When it runs, the language would generate
+a graph image and then copy it to the clipboard for the user to paste or
+ideally make it appear in place, but this might be difficult. An early issue I 
+could forsee happening is if a user specifies a curve that is not supported in
+the program.  For example, if they specified `ProductionFunction()` and we didn't
+support that curve, there would be an error.  The way for this error to appear
+would likely be through some sort of linter, much like other programming
+languages have. We would then notify users at compile time, or in this case on
+save that there is an issue and communicate it clearly.
 
 
 ### Expressiveness
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
-
+It should be easy to both take notes in natural language and generate graphs in
+the same editor or workspace. It should be possible to create extremely custom
+graphs, but I imagine that it would be difficult and may require more time
+than available during a class session.  It should be impossible to do anything
```

Comment:
Worst case you should be able to embed an image, as well as specify arbitrary points (i.e. scatter-plot style).

If you wanted to do computer vision, have the runtime analyze the graph and generate the appropriate function calls... but that seems hard.